napari/utils/events/event_utils.py

```diff
@@ -19,7 +35,25 @@ def disconnect_events(emitter, listener):
             # will be kept. If *callback* is a callable object then it is
             # not attached to the listener and does not need to be
             # disconnected
-            if isinstance(callback, tuple) and callback[0] is weakref.ref(
-                listener
-            ):
+            if isinstance(callback, tuple) and callback[0] is weak_listener:
                 em.disconnect(callback)
+
+
+def connect_setattr(emitter: Emitter, obj, attr: str):
+    ref = weakref.ref(obj)
+
+    def _cb(*value):
+        setattr(ref(), attr, value)
+
+    emitter.connect(_cb)
+    # weakref.finalize(obj, emitter.disconnect, _cb)
```

Comment:
maybe just leave a note here as to why these are commented (or delete them).  they're very tempting otherwise for future tinkerers :)

Reply:
I add an explanation for the future development. Could You check it?

Reply:
looks good thanks!

Reply:
looks good thanks!

Reply:
looks good thanks!